File: triatlon-cs/Repository/DB/AthleteDbRepository.cs
using System.Collections.Generic;
using System.Data;
using log4net;
using Model.Person;
using Repository.Utils;

namespace Repository.DB
{
    public class AthleteDbRepository : IAthleteRepository
{
    
    // Private Properties

    private static readonly ILog Logger = LogManager.GetLogger("AthleteDbRepository");
    private readonly IDictionary<string, string> _properties;
    
    // Lifecycle

    public AthleteDbRepository(IDictionary<string, string> properties)
    {
        Logger.Info("Creating AthleteDbRepository");
        _properties = properties;
    }

    // Repository Methods
    
    public Athlete FindOne(int id)
    {
        Logger.InfoFormat("Entering FindOne with value {0}", id);
        IDbConnection connection = DbUtils.GetConnection(_properties);
        using (var command = connection.CreateCommand())
        {
            command.CommandText = "select * from athletes where id=@id";
            var parameterId = command.CreateParameter();
            parameterId.ParameterName = "@id";
            parameterId.Value = id;
            command.Parameters.Add(parameterId);
            using (var dataReader = command.ExecuteReader())
            {
                if (dataReader.Read())
                {
                    int _id = dataReader.GetInt32(0);
                    string firstName = dataReader.GetString(1);
                    string lastName = dataReader.GetString(2);
                    var athlete = new Athlete(firstName, lastName);
                    athlete.Id = _id;
                    Logger.InfoFormat("Exiting FindOne with value {0}", athlete);
                    return athlete;
                }
            }
        }
        Logger.InfoFormat("Exiting FindOne with value {0}", null);
        return null;
    }

    public IEnumerable<Athlete> FindAll()
    {
        Logger.Info("Entering FindAll");
        IDbConnection connection = DbUtils.GetConnection(_properties);
        IList<Athlete> athletes = new List<Athlete>();
        using (var command = connection.CreateCommand())
        {
            command.CommandText = "select * from athletes";
            using (var dataReader = command.ExecuteReader())
            {
                while (dataReader.Read())
                {
                    int _id = dataReader.GetInt32(0);
                    string firstName = dataReader.GetString(1);
                    string lastName = dataReader.GetString(2);
                    var athlete = new Athlete(firstName, lastName);
                    athlete.Id = _id;
                    athletes.Add(athlete);
                }
            }
        }
        Logger.Info("Exiting FindAll");
        return athletes;
    }

    public Athlete Save(Athlete entity)
    {
        Logger.InfoFormat("Entering Save with value {0}", entity);
        IDbConnection connection = DbUtils.GetConnection(_properties);
        using (var command = connection.CreateCommand())
        {
            command.CommandText = "insert into athletes (firstName, lastName) values (@firstName, @lastName)";
            var firstNameParameter = command.CreateParameter();
            firstNameParameter.ParameterName = "@firstName";
            firstNameParameter.Value = entity.FirstName;
            command.Parameters.Add(firstNameParameter);
            var lastNameParameter = command.CreateParameter();
            lastNameParameter.ParameterName = "@lastName";
            lastNameParameter.Value = entity.LastName;
            command.Parameters.Add(lastNameParameter);
            var result = command.ExecuteNonQuery();
            Logger.InfoFormat("Exiting Save with value {0}", result);
        }
        return entity;
    }

    public int Delete(int id)
    {
        Logger.InfoFormat("Entering Delete with value {0}", id);
        IDbConnection connection = DbUtils.GetConnection(_properties);
        using (var command = connection.CreateCommand())
        {
            command.CommandText = "delete from athletes where id=@id";
            var idParameter = command.CreateParameter();
            idParameter.ParameterName = "@id";
            idParameter.Value = id;
            command.Parameters.Add(idParameter);
            var result = command.ExecuteNonQuery();
            Logger.InfoFormat("Exiting Delete with value {0}", result);
        }
        return id;
    }

    public Athlete Update(Athlete entity)
    {
        Logger.InfoFormat("Entering Update with value {0}", entity);
        IDbConnection connection = DbUtils.GetConnection(_properties);
        using (var command = connection.CreateCommand())
        {
            command.CommandText = "update athletes set firstName=@firstName, lastName=@lastName where id=@id";
            var idParameter = command.CreateParameter();
            idParameter.ParameterName = "@id";
            idParameter.Value = entity.Id;
            command.Parameters.Add(idParameter);
            var firstNameParameter = command.CreateParameter();
            firstNameParameter.ParameterName = "@firstName";
            firstNameParameter.Value = entity.FirstName;
            command.Parameters.Add(firstNameParameter);
            var lastNameParameter = command.CreateParameter();
            lastNameParameter.ParameterName = "@lastName";
            lastNameParameter.Value = entity.LastName;
            command.Parameters.Add(lastNameParameter);
            var result = command.ExecuteNonQuery();
            Logger.InfoFormat("Exiting Update with value {0}", result);
        }
        return entity;
    }
}
}

File: triatlon-java/src/main/java/ro/ubbcluj/scs/dfar2166/triatlonjava/repository/db/AthleteDBRepository.java
package ro.ubbcluj.scs.dfar2166.triatlonjava.repository.db;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ro.ubbcluj.scs.dfar2166.triatlonjava.model.person.Athlete;
import ro.ubbcluj.scs.dfar2166.triatlonjava.repository.AthleteRepository;
import ro.ubbcluj.scs.dfar2166.triatlonjava.utils.JDBCUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

// AthleteDBRepository

public class AthleteDBRepository implements AthleteRepository {

    // Private Properties

    private final JDBCUtils databaseUtils;

    // Private Class Properties

    private static final Logger logger = LogManager.getLogger();

    // Lifecycle

    public AthleteDBRepository(Properties properties) {
        logger.info("Initialising AthleteDBRepository with properties: {}", properties);
        databaseUtils = new JDBCUtils(properties);
    }

    // Override Methods (Repository)

    @Override
    public Athlete findOne(Integer integer) {
        logger.traceEntry("Finding task {}", integer);
        Connection connection = databaseUtils.getConnection();
        Athlete athlete = null;
        try (PreparedStatement preparedStatement = connection.prepareStatement("select * from athletes where id=?")) {
            preparedStatement.setInt(1, integer);
            try (ResultSet resultSet = preparedStatement.executeQuery()) {
                while (resultSet.next()) {
                    var id = resultSet.getInt("id");
                    var firstName = resultSet.getString("firstName");
                    var lastName = resultSet.getString("lastName");
                    athlete = new Athlete(firstName, lastName);
                    athlete.setId(id);
                }
                logger.traceExit("Found 1 instance");
            }
        } catch (SQLException ex) {
            logger.error(ex);
        }
        logger.traceExit();
        return athlete;
    }

    @Override
    public Iterable<Athlete> findAll() {
        logger.traceEntry("Finding task");
        Connection connection = databaseUtils.getConnection();
        List<Athlete> athletes = new ArrayList<>();
        try (PreparedStatement preparedStatement = connection.prepareStatement("select * from athletes")) {
            try (ResultSet resultSet = preparedStatement.executeQuery()) {
                while (resultSet.next()) {
                    var id = resultSet.getInt("id");
                    var firstName = resultSet.getString("firstName");
                    var lastName = resultSet.getString("lastName");
                    Athlete athlete = new Athlete(firstName, lastName);
                    athlete.setId(id);
                    athletes.add(athlete);
                }
                logger.traceExit("Found {} instances", athletes.size());
            }
        } catch (SQLException ex) {
            logger.error(ex);
        }
        logger.traceExit();
        return athletes;
    }

    @Override
    public Athlete save(Athlete entity) {
        logger.traceEntry("Saving task {}", entity);
        Connection connection = databaseUtils.getConnection();
        try (PreparedStatement preparedStatement = connection.prepareStatement("insert into athletes (firstName, lastName) values (?, ?)")) {
            preparedStatement.setString(1, entity.getFirstName());
            preparedStatement.setString(2, entity.getLastName());
            int result = preparedStatement.executeUpdate();
            logger.traceExit("Saved {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
        }
        logger.traceExit();
        return entity;
    }

    @Override
    public Integer delete(Integer integer) {
        logger.traceEntry("Deleting task {}", integer);
        Connection connection = databaseUtils.getConnection();
        try (PreparedStatement preparedStatement = connection.prepareStatement("delete from athletes where id=?")) {
            preparedStatement.setInt(1, integer);
            int result = preparedStatement.executeUpdate();
            logger.traceExit("Deleted {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
        }
        logger.traceExit();
        return integer;
    }

    @Override
    public Athlete update(Athlete entity) {
        logger.traceEntry("Updating task {}", entity);
        Connection connection = databaseUtils.getConnection();
        try (PreparedStatement preparedStatement = connection.prepareStatement("update athletes set firstName=?, lastName=? where id=?")) {
            preparedStatement.setString(1, entity.getFirstName());
            preparedStatement.setString(2, entity.getLastName());
            preparedStatement.setInt(3, entity.getId());
            int result = preparedStatement.executeUpdate();
            logger.traceExit("Updated {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
        }
        logger.traceExit();
        return entity;
    }

}
